dll/win32/shlwapi/propbag.cpp

```diff
@@ -1806,3 +1806,27 @@ EXTERN_C VOID FreeViewStatePropertyBagCache(VOID)
     g_pCachedBag.Release();
     ::LeaveCriticalSection(&g_csBagCacheLock);
 }
+
+/**************************************************************************
+ *  SHGetPerScreenResName (SHLWAPI.533)
+ *
+ * @see https://www.geoffchappell.com/studies/windows/shell/shlwapi/api/propbag/getperscreenresname.htm
+ */
+INT WINAPI
+SHGetPerScreenResName(
+    _Out_writes_(cchBuffer) LPWSTR pszBuffer,
+    _In_ INT cchBuffer,
+    _In_ DWORD dwReserved)
+{
+    if (dwReserved)
+        return 0;
+
+    INT cxWidth = ::GetSystemMetrics(SM_CXFULLSCREEN);
+    INT cyHeight = ::GetSystemMetrics(SM_CYFULLSCREEN);
+    INT cMonitors = ::GetSystemMetrics(SM_CMONITORS);
+    INT ret = wnsprintfW(pszBuffer, cchBuffer, L"%dx%d(%d)", cxWidth, cyHeight, cMonitors);
```

Comment:
Why don't you use `StringCchPrintfW` instead? Thought we actually discouraged the use of these unsafe functions...

Reply:
Fixed.